### PR TITLE
[ etherIntegrate ] : smart contract integration and dependency patches

### PR DESCRIPTION
**Pull request by skywall34**

### CODE REVIEW

File: smart_contracts/contracts/DataAuth.sol [hunk 1]
```diff
@@ -109,8 +108,6 @@ contract DataAuth is Ownable{
      */
     function editUser(bytes32 _lockId, string memory _username, string memory _group, uint32[] memory _databaseList) public user_exists(_lockId) onlyOwner returns (bool){
 
-        if (owns_user(_lockId) == false)
```
Comment (huynhtastic) [inline smart_contracts/contracts/DataAuth.sol]: The function definition doesn't include an owns_user check. Is this valid?

File: smart_contracts/truffle-config.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ module.exports = {
     //
      development: {
       host: "127.0.0.1",     // Localhost (default: none)
-      port: 7545,            // Standard Ethereum port (default: none)
+      port: 8545,            // Standard Ethereum port (default: none)
```
Comment (huynhtastic) [inline smart_contracts/truffle-config.js]: Is there a reason why you changed the port #?

File: package.json [hunk 3]
```diff
@@ -27,7 +27,11 @@
   "homepage": "https://github.com/skywall34/DataAuth#readme",
   "dependencies": {
     "@babel/runtime": "^7.4.2",
+    "babel": "^6.23.0",
```
[the remark this answers is not in the document]
Comment (skywall34) [reply package.json]: updated, see the latest commit

File: package.json [hunk 4]
```diff
@@ -27,7 +27,11 @@
...
+    "babel-cli": "^6.26.0",
```
[the remark this answers is not in the document]
Comment (skywall34) [reply package.json]: updated, see the latest commit

File: package.json [hunk 5]
```diff
@@ -27,7 +27,11 @@
...
+    "babel-preset-es2015": "^6.24.1",
```
[the remark this answers is not in the document]
Comment (skywall34) [reply package.json]: updated, see the lates commit

File: smart_contracts/truffle-config.js (hunk 2 above)
Comment (skywall34) [reply smart_contracts/truffle-config.js]: ganache-cli runs on 8454 on default. If you run ganache as an application and use the ui however, it runs on 7545. It really depends on what you're developing on.

File: smart_contracts/contracts/DataAuth.sol (hunk 1 above)
Comment (skywall34) [reply smart_contracts/contracts/DataAuth.sol]: owns_user isn't needed since user_exists modifier has already been defined.

Comment (huynhtastic) [reply smart_contracts/contracts/DataAuth.sol]: Ahh didn't see that at the end of the function def.

**Comment by huynhtastic** [comment]
Rebased these commits

**Pull request merged**